Issue 27 bottom navigator component

ErikaKK opened the pull request:
## Change Summary
Usage: 
- `BottomNav` component  is fixed at the bottom, with `width:100%`. 
- onClick color change 
- Can add `link` property to list `iconLists` to link the item divs to other pages

### Change Form
**Fill this up (NA if not available). If a certain criteria is not met, can you please give a reason.**

- [x] The pull request title has an issue number
- [ ] The change works by "Smoke testing" or quick testing
- [ ] The change has tests
- [ ] The change has documentation

## Other Information
**[Is there anything in particular in the review that I should be aware of?]**
Font is not defined

# Related issue

- Resolve #27

# Related issue

- Resolve #27

ErikaKK committed:
Add bottom navigation icons svg files. Add bottomNav component using tailwind font size of 12px, height of 80px

ErikaKK committed:
Delete unused component

ErikaKK committed:
Add index in map method

ErikaKK committed:
Add svg icon component. Change icon color and layout size to become the same as Figma design

ErikaKK committed:
Add interface and comment

ErikaKK committed:
Add onClick color change. Make bottomNav style fix at the bottom

yunho7687 commented:
I have updated the main branch. Please update your branch to resolve the flake8 errors.
Please review my commit to see the changes I have made.

SodaVolcano commented on client/src/components/ui/bottomNav.tsx:
```diff
@@ -0,0 +1,140 @@
+  };
+
+  return (
+    <div className="border-t-penni-border-light-mode fixed bottom-0 h-[83px] w-full border-t-2">
```
`h-[83px]` consider replacing with closest matching tailwind class

SodaVolcano commented on client/src/components/ui/bottomNav.tsx:
```diff
@@ -0,0 +1,140 @@
+
+  return (
+    <div className="border-t-penni-border-light-mode fixed bottom-0 h-[83px] w-full border-t-2">
+      <ul className="flex h-[49px] cursor-pointer text-[10px] leading-3">
```
Same for `h-[49px]`, `text-[10px]`

SodaVolcano commented on client/src/components/ui/bottomNav.tsx:
```diff
@@ -0,0 +1,140 @@
+        {iconLists.map((iconItem, index) => (
+          <li
+            key={index}
+            className={`${changeColor === index ? "text-penni-main" : "text-penni-text-tertiary-light-mode"} w-1/3`}
```
Does template string `$(changeColor)...` actually work? Tailwind with just-in-time compilation can't parse string concatenation and also template strings when I was making the other components. If this doesn't work, add `"text-penni-main", "text-penni-text-tertiary..."` to `safelist` in tailwind config file.

HuxleyBerry replied:
It should work since the class names are written out in full. See https://tailwindcss.com/docs/content-configuration#dynamic-class-names

SodaVolcano commented on client/src/components/ui/bottomNav.tsx:
```diff
@@ -0,0 +1,140 @@
+          >
+            <Link
+              href={iconItem.link || ""}
+              className="flex flex-col items-center pt-[6px]"
```
Same for `pt-[6px]`

SodaVolcano reviewed (changes requested):
Also if the `.svg` files in `/public/img` is not used consider removing them. Overall pretty good, with some minor changes requested.

SodaVolcano commented on client/src/components/ui/bottomNav.tsx:
```diff
@@ -0,0 +1,140 @@
+  );
+};
+
+export { BottomNavIcon1, BottomNavIcon2, BottomNavIcon3 };
```
Consider moving all `BottomNavIcon` components and interfaces to `/components/icons.tsx`, and rename from `BottomNavIcon` to something more descriptive (describe the actual icons).

UPDATE: Corrected lines of code being highlighted

SodaVolcano commented on client/src/components/ui/bottomNav.tsx:
```diff
@@ -0,0 +1,140 @@
+              </div>
+              <p>{iconItem.text}</p>
+            </Link>
+          </li>
```
Maybe move the `<li>` element inside `map()` to its own component to avoid deep nesting? Ehh up to you, the code is fine as is so might not be worth the hassle.

UPDATE: Corrected lines of code being highlighted

SodaVolcano reviewed (changes requested):
Corrected lines of code being highlighted

ErikaKK committed:
Merge branch 'main' into issue-27-Bottom_Navigator_Component

ErikaKK committed:
Remove unused svg file. Move icon components to icons.tsx

Make change to BottomNav, will require 1 prop of the index of icon. change tailwind  class

ErikaKK commented:
Font can be changed after #42 is solved

ErikaKK committed:
Rename the file to lowercase. Change  icon names  in icons.tsx
